client/my-sites/comments/comment-navigation/index.jsx

```diff
@@ -65,6 +66,19 @@ export class CommentNavigation extends Component {
 		}
 	};
 
+	emptyPermanently = () => {
+		const { status, translate } = this.props;
+		if (
+			window?.confirm(
```

Comment:
Why is the `window` optional chaining necessary? Does that run in SSR context?

Reply:
I was being cautious because the equivalent function for bulk deleting comments in the same component does this:

```
if (
	typeof window === 'undefined' ||
	window.confirm( translate( 'Delete these comments permanently?' ) )
) {
```

What's the best way to find out if a section runs in the SSR context?

I wonder if that code was accounting for some other situation where `window` wasn't available - tests perhaps?

Reply:
FWIW I'm fine with keeping the optional chaining, especially if there's a good reason for it.

> What's the best way to find out if a section runs in the SSR context?

Basically, tracing the `CommentNavigation` component up to its corresponding section:

https://github.com/Automattic/wp-calypso/blob/bb00cff9a4e21a8beed8cfa95c17e09bf428bda4/client/sections.js#L391-L396

and checking if `isomorphic` is set to `true` for that section. If it's not then, it doesn't support SSR.

> I wonder if that code was accounting for some other situation where window wasn't available - tests perhaps?

Perhaps that's why! In any case, we should be able to mock what's missing, or enable `jsdom` testing environment for the tests, which supports most of the browser APIs.

Reply:
Thanks for filling me in on `isomorphic: true` :) Doesn't seem like there's a compelling reason to keep it. Removed in [260f745](https://github.com/Automattic/wp-calypso/pull/74315/commits/260f745ebb798039c11c56950fe501638c48d0c2).